cmake/onnxruntime_framework.cmake

```diff
@@ -36,6 +36,99 @@ endif()
 source_group(TREE ${REPO_ROOT} FILES ${onnxruntime_framework_srcs})
 
 onnxruntime_add_static_library(onnxruntime_framework ${onnxruntime_framework_srcs})
+
+if (onnxruntime_USE_CLOUD)
+
+  include(ExternalProject)
+
+  if (WIN32)
+
+    function(get_vcpkg)
+      ExternalProject_Add(vcpkg
+        GIT_REPOSITORY https://github.com/microsoft/vcpkg.git
+        PREFIX vcpkg
+        CONFIGURE_COMMAND ""
+        INSTALL_COMMAND ""
+        UPDATE_COMMAND ""
+        BUILD_COMMAND "<SOURCE_DIR>/bootstrap-vcpkg.bat")
+
+      ExternalProject_Get_Property(vcpkg SOURCE_DIR)
+      set(VCPKG_SRC ${SOURCE_DIR} PARENT_SCOPE)
+      set(VCPKG_DEPENDENCIES "vcpkg" PARENT_SCOPE)
+    endfunction()
+
+    function(vcpkg_install PACKAGE_NAME)
+      add_custom_command(
+        OUTPUT ${VCPKG_SRC}/packages/${PACKAGE_NAME}_x64-windows/BUILD_INFO
+        COMMAND ${VCPKG_SRC}/vcpkg install ${PACKAGE_NAME}:x64-windows
+        WORKING_DIRECTORY ${VCPKG_SRC}
+        DEPENDS vcpkg)
+
+      add_custom_target(get${PACKAGE_NAME}
+        ALL
+        DEPENDS ${VCPKG_SRC}/packages/${PACKAGE_NAME}_x64-windows/BUILD_INFO)
+
+      list(APPEND VCPKG_DEPENDENCIES "get${PACKAGE_NAME}")
+      set(VCPKG_DEPENDENCIES ${VCPKG_DEPENDENCIES} PARENT_SCOPE)
+    endfunction()
+
+    get_vcpkg()
+    vcpkg_install(openssl)
+    vcpkg_install(openssl-windows)
+    vcpkg_install(rapidjson)
+    vcpkg_install(re2)
+    vcpkg_install(boost-interprocess)
+    vcpkg_install(boost-stacktrace)
+    vcpkg_install(zlib)
+    vcpkg_install(pthread)
+    vcpkg_install(b64)
+
+    ExternalProject_Add(triton
```

Comment:
>ExternalProject_Add [](http://example.com/codeflow?start=4&length=19)

Triton introduced dependency on zlib dll on windows, we must call out aloud on this.